include/splinterdb/splinterdb.h

```diff
@@ -72,6 +74,8 @@ typedef struct {
    uint64 max_branches_per_node;
    uint64 use_stats;
    uint64 reclaim_threshold;
+   uint8  num_bg_threads[SPLINTERDB_NUM_TASK_TYPES]; // = {0} => no bg threads
```

Comment:
Also this doesn't make clear which type is which and the enum is defined in `task.c`, so it's not straightforward to translate.

What I think we should do is have `num_memtable_bg_threads` and `num_normal_bg_threads` here and get rid of `SplinterDB_NUM_TASK_TYPES`.

I don't really like having the application provide this split to be honest. I think that the application should provide a `max_bg_threads` parameter and `task.c` should handle the load balancing.

Reply:
I second that we need to either define two separate `bg_threads` fields or at least expose the `#define`s for the different classes so that users can use those to set the classes.

Also 100% agree this should all be automated by task.c so the user just has to specify the max total bg threads (or maybe even just whether to use bg threads or not).  But that's probably out of scope for your PR.

Reply:
@ajhconway , @rtjohnso - Based on what we discussed in review meeting that this PR does not have to reconcile the interfaces and that we can get this 'enablement feature' in first, here is what I did to address your comments:

1. Externalized two new fields, `num_memtable_bg_threads` and `num_normal_bg_threads` to splinter_config{}.
2. Interface to `task_system_create()` remains unchanged. It will receive `use_bg_threads == TRUE` only if both fields above are non-zero. (This addresses one of the concerns that the system will be unstable if we configure only one of these task-types with bg-threads count.)
3. Config options are now renamed symmetrically to be `--num-normal-bg-threads` and `--num-memtable-bg-threads`.
4. Added checks in `config_parse()` to enforce that both config params are == 0 or both are != 0. (Pulled-in similar checks Rob added in [PR 497, here](https://github.com/vmware/splinterdb/pull/497/files#diff-b74ae5c4f955f7aaf2dabd10a7a4cdbc03e7afff831220370151d6b01b05ce43R345).)
5. Amended cases in `task_system_test.c` and calls in `test.sh` to invoke this test-case with and w/o background threads configured.

These changes should wholesale address the overall issues raised in this comment-stream.